### DISCUSSION TITLE
insert bibtex field at point (or multiple templates for `citar-insert-reference`)

### DISCUSSION THREAD

**Discussion by tabroughton** [emacs-citar/citar#774] [Ideas]
I would like to be able to insert a field from a reference in `.bib` file into my buffer at point. 

I'm not sure if this is possible with `citar`  or `org-cite` atm, I'm assuming not. 

I see it as being a possible extension to the functionality of `citar-insert-reference` but instead of only using the `preview` template from `citar-templates`  one would be able to add the the list of `citar-templates` and when using `citar-insert-reference` (or similar command) the template could be selected.  -  that's solution more than requirement, I don't really mind what the solution might be and if it is already possible I'd be interested to know how.

Reason I am thinking of this is I am developing a workflow for reasearch etc and I am using zotero.  My use case at the moment is something along the lines of:

1. Find resource and import into zotero
2. Read pdf in zotero and make annotations
3. export annotations into a note in zotero and use Better Bibtex plugin to export notes in .bib file
4. I would like to be able to import those notes into an existing org-roam file (or any existing buffer).
5. Ideally I'd have a parser that can interpret the latex in the field (the exported annotation is latex in the `note` field) and separate out notes to select from, but I feel this would be the next step

because I'm just starting on my toolchain/workflow for academic research I'd also be interested in how other people manage their annotations whilst reading a resource and whether my idea here is already accommodated elsewhere or if there is alternative approach that might work for me.

**Comment by bdarcus** [comment]
Seems like you started with a solution rather than the problem you want to solve ;-)

It sounds like really what you want is to be able to import Zotero annotations into `org-roam`, ideally pretty seamlessly? 

See #685 for a related issue, which is more about integrating citar and Zotero annotations. If that worked well, you could just keep the annotations in Zotero maybe?

If not, I wonder if `org-noter` is the better target, at least ideally? Not that it and `org-roam` are mutually-exclusive though.

As for your implementation question, that sort of thing is easy to do with a new command that wraps `citar-get-value`. 

For example:

```elisp
(defun ex/insert-field ()
  (interactive)
  (let* ((field (completing-read "Field: " '("author" "editor" "note" "title")))
         (citekey (citar-select-ref)))
    (insert (citar-get-value field citekey))))
```

**Reply by bdarcus** [reply]
[This](https://org-roam.discourse.group/t/new-zotero-pdf-reader-and-org-roam-org-roam-bibtex-workflow/1364) says Better BibTeX exports these as LaTeX notes.

Here's an (untested) example adapted from https://github.com/org-roam/org-roam-bibtex/issues/178#issuecomment-798898300 that _should_  work, to convert those notes to org.

```elisp
(defun ex/latex-note-to-org (citekey)
  "Convert Better BibTeX exported LaTeX notes to org." ; Zotero annotations
  (let* ((note (citar-get-value "note" citekey))
         (pandoc-command "pandoc --from latex --to org")
         result)
    (with-temp-buffer
      (shell-command (format "echo \"%s\" | %s" note pandoc-command)
                     (current-buffer))
      (setq result (buffer-substring-no-properties (point-min) (point-max))))))
```

 Just need to figure out how to hook it up with some specific output (insert into buffer, integrate into capture template, etc.).

Effectively, this is a sort of filter run on some content. It may be possible to adapt something like this to the new `citar-org-roam` capture template support.

I created an issue for this at https://github.com/emacs-citar/citar-org-roam/issues/43.
